bundler/lib/bundler/retry.rb

```diff
@@ -50,7 +50,7 @@ def fail_attempt(e)
       end
       return true unless name
       Bundler.ui.info "" unless Bundler.ui.debug? # Add new line in case dots preceded this
-      Bundler.ui.warn "Retrying #{name} due to error (#{current_run.next}/#{total_runs}): #{e.class} #{e.message}", Bundler.ui.debug?
+      Bundler.ui.warn "Retrying #{name} due to error (#{current_run.next}/#{total_runs}): #{e.class} #{e.message}", true
```

Comment:
Now that the `newline` parameter to `warn` is being correctly followed ([see comment](https://github.com/rubygems/rubygems/pull/7957/files#r1726190085)), this warning statement no longer worked as intended. When `Bundler.ui.debug?` was false, it was causing newlines to be disabled. I fixed this by explicitly passing `true` so that newlines are always printed. The default behavior will omit the newline if the message string ends with a space character, which could happen if `e.message` is nil or empty.